src/main/java/com/user_service/controllers/UserAuthenticationController.java

```diff
@@ -16,67 +27,32 @@ public class UserAuthenticationController {
     private UserAuthenticationService userAuthenticationService;
 
     @PostMapping(value = "/register")
-    public ResponseEntity<String> signUpUser(@RequestBody SignUpUserDto signUpUserDto) {
-        SignUpResponseDto signUpResponseDto = userAuthenticationService.signUpUser(signUpUserDto);
-
-        if (signUpResponseDto.getSignUpResponse() == null) {
-            return new ResponseEntity<>(signUpResponseDto.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
-        }
-        return new ResponseEntity<>("Account sign up, confirmation code sent to " + signUpUserDto.getEmail(), HttpStatus.OK);
-    }
-
-    @PostMapping(value = "/confirm")
-    public ResponseEntity<String> confirmSignUp(@RequestBody ConfirmRegisterDto confirmRegisterDto) {
-        ConfirmSignUpResponseDto confirmSignUpResponseDto = userAuthenticationService.confirmSignUp(confirmRegisterDto);
-
-        if (confirmSignUpResponseDto.getConfirmSignUpResponse() == null) {
-            return new ResponseEntity<>(confirmSignUpResponseDto.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
-        }
-        return new ResponseEntity<>("Account confirmed successfully", HttpStatus.OK);
+    public ResponseEntity<SignUpResponseDto> signUpUser(@RequestBody SignUpUserDto signUpUserDto) {
```

Comment:
I think Spring should be able to return from the function just `SignUpResponseDto` and Spring will do the translation to the `ResponseEntity`.

And if you want to return sth different than 200, you can use `@ResponseStatus(HttpStatus.CREATED)`.

It's a minor thing, but in the long run it saves a bit of code in most cases where you return just 200.